File: Backend/connecthub/src/main/java/com/example/connecthub/service/CommentService.java
package com.example.connecthub.service;

import com.example.connecthub.dto.request.CreateCommentRequest;
import com.example.connecthub.dto.request.UpdateCommentRequest;
import com.example.connecthub.dto.response.CommentResponse;

import java.util.List;


public interface CommentService {

    CommentResponse createComment(
            Long postId,
            String email,
            CreateCommentRequest request
    );

    List<CommentResponse> getCommentsByPost(Long postId);

    CommentResponse updateComment(
            Long id,
            String email,
            UpdateCommentRequest request
    );

    void deleteComment(
            Long id,
            String email
    );
}

File: frontend/src/services/commentService.js
import client from '../api/client';

export const commentService = {
  getCommentsByPost: async (postId) => {
    const response = await client.get(`/posts/${postId}/comments`);
    return response.data; // List<CommentResponse>
  },

  createComment: async (postId, content) => {
    const response = await client.post(`/posts/${postId}/comments`, { content });
    return response.data; // CommentResponse
  },

  updateComment: async (commentId, content) => {
    const response = await client.put(`/comments/${commentId}`, { content });
    return response.data; // CommentResponse
  },

  deleteComment: async (commentId) => {
    await client.delete(`/comments/${commentId}`);
  },
};
